indra/newview/llviewermessage.cpp

```diff
@@ -2749,6 +2749,11 @@ void process_chat_from_simulator(LLMessageSystem *msg, void **user_data)
             msg_notify["from_id"] = chat.mFromID;
             msg_notify["source_type"] = chat.mSourceType;
             on_new_message(msg_notify);
+
+
+            msg_notify["chat_type"] = chat.mChatType;
+            msg_notify["message"] = mesg;
+            LLEventPumps::instance().obtain("LLNearbyChat").post(msg_notify);
```

Comment:
Is one of these fields the chat channel? That seems useful for a listener to know.

Reply:
As far as I know, only channel 0 messages are sent to the avatars. All other channels can only be listened by lsl scripts.